// react-bootstrap-spinner/dist/index.es.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';

var classCallCheck = function (instance, Constructor) {
  if (!(instance instanceof Constructor)) {
    throw new TypeError("Cannot call a class as a function");
  }
};

var createClass = function () {
  function defineProperties(target, props) {
    for (var i = 0; i < props.length; i++) {
      var descriptor = props[i];
      descriptor.enumerable = descriptor.enumerable || false;
      descriptor.configurable = true;
      if ("value" in descriptor) descriptor.writable = true;
      Object.defineProperty(target, descriptor.key, descriptor);
    }
  }

  return function (Constructor, protoProps, staticProps) {
    if (protoProps) defineProperties(Constructor.prototype, protoProps);
    if (staticProps) defineProperties(Constructor, staticProps);
    return Constructor;
  };
}();

var inherits = function (subClass, superClass) {
  if (typeof superClass !== "function" && superClass !== null) {
    throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
  }

  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true
    }
  });
  if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
};

var possibleConstructorReturn = function (self, call) {
  if (!self) {
    throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  }

  return call && (typeof call === "object" || typeof call === "function") ? call : self;
};

var ReactBootstrapSpinner = function (_Component) {
  inherits(ReactBootstrapSpinner, _Component);

  function ReactBootstrapSpinner() {
    classCallCheck(this, ReactBootstrapSpinner);
    return possibleConstructorReturn(this, (ReactBootstrapSpinner.__proto__ || Object.getPrototypeOf(ReactBootstrapSpinner)).apply(this, arguments));
  }

  createClass(ReactBootstrapSpinner, [{
    key: 'render',
    value: function render() {
      var type = this.props.type;

      var divStyle = {
        height: this.props.size + 'rem',
        width: this.props.size + 'rem'
      };

      var color = this.props.color;

      var SpinnerClass = "spinner-" + type + ' ' + "text-" + color;

      return React.createElement(
        'div',
        { className: SpinnerClass, style: divStyle, role: 'status' },
        React.createElement(
          'span',
          { className: 'sr-only' },
          'Loading...'
        )
      );
    }
  }]);
  return ReactBootstrapSpinner;
}(Component);

ReactBootstrapSpinner.propTypes = {
  size: PropTypes.string,
  type: PropTypes.string,
  color: PropTypes.string
};

export default ReactBootstrapSpinner;
